azkaban-spi/src/main/java/com/linkedin/azkaban/spi/AzkabanException.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2017 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package com.linkedin.azkaban.spi;
+
+public class AzkabanException extends RuntimeException {
```

Comment:
This is kind of a debate and I don't have a strong opinion on this.

IMO Checked exceptions add more overhead in the source code than handle the actual problem. Normally people just bubble it up by wrapping it into a `RuntimeException`. I wanted to keep it simple. Also, this allows to not handle the exception if they are not capable of handling it.